R/use-standalone.R

```diff
@@ -38,6 +38,10 @@ use_standalone <- function(repo_spec, file = NULL, ref = NULL, host = NULL) {
   dest_path <- path("R", paste0("import-", file))
 
   lines <- read_github_file(repo_spec, path = src_path, ref = ref, host = host)
+  if (is.na(lines)) {
+    ui_oops("Failed to retrieve content from {ui_path(src_path)}")
+    rlang::cnd_signal(attr(lines, "error"))
+  }
   lines <- c(standalone_header(repo_spec, src_path), lines)
```

Comment:
This isn't exactly a canonical way to rethrow the error we got from gh, but it gets the job done. Better ideas? Here's how it looks (the backtrace is displayed more nicely in an interactive session):

``` r
devtools::load_all("~/rrr/usethis")
#> ℹ Loading usethis
use_github_file("r-lib/usethis", path = "NOPE")
#> ✔ Saving 'r-lib/usethis/NOPE' to 'NOPE'
#> ✖ Failed to retrieve content from 'NOPE'
#> Error in `gh_process_response()`:
#> ! GitHub API error (404): Not Found
#> ✖ URL not found: <https://api.github.com/repos/r-lib/usethis/contents/NOPE>
#> ℹ Read more at
#>   <https://docs.github.com/rest/reference/repos#get-repository-content>

#> Backtrace:
#>      ▆
#>   1. └─usethis::use_github_file("r-lib/usethis", path = "NOPE")
#>   2.   └─usethis:::read_github_file(...) at usethis/R/use_github_file.R:73:2
#>   3.     └─(purrr::safely(gh::gh))(...) at usethis/R/use_github_file.R:109:2
#>   4.       ├─purrr:::capture_error(.f(...), otherwise, quiet)
#>   5.       │ └─base::tryCatch(...)
#>   6.       │   └─base (local) tryCatchList(expr, classes, parentenv, handlers)
#>   7.       │     └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
#>   8.       │       └─base (local) doTryCatch(return(expr), name, parentenv, handler)
#>   9.       └─gh (local) .f(...)
#>  10.         └─gh:::gh_process_response(raw)
#>  11.           └─gh:::gh_error(response)
#>  12.             └─cli::cli_abort(...)
#>  13.               └─rlang::abort(...)
```

<sup>Created on 2023-02-23 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>